main.js

```diff
@@ -367,126 +367,105 @@ function readValue(port) {
     });
 }
 
-function syncPort(port, data, callback) {
+// syncPort waits for everything it calls, so upon return, when the GPIO
+// handlers, etc. are setup their states are guaranteed to be in place.
+
+// Our own deleteState that logs an error only if it's not, Not Exists
+async function deleteState(stateName) {
+    try {
+        await adapter.delObjectAsync(stateName);
+    } catch (err) {
+        if (err != 'Error: Not exists') {
+            throw new Error(`Failed to delete object ${stateName}: ${err}`);
+        }
+    }
+    try {
+        await adapter.delStateAsync(stateName);
```

Comment:
There's a comment next to `delState` that reads:

> Deletes a state from the states DB, but not the associated object. Consider using @link{deleteState} instead

However, I did consider - and even try ;) - using `deleteState` but it does not remove the object. I probably got the wrong end of the stick there? Anyhow, so have kept the two separate delObject/State calls. Please say if this can be done in one operation.

Reply:
delState ist just remving the state ... delObject is removing the object

Reply:
Why does it say, "Consider using @link{deleteState} instead" then?

Reply:
deleteState:         deleteState = (parentDevice, parentChannel, stateName, options, callback) => {

delState: delState = (id, options, callback) => {

clearer now? differnt usecases. deeteState works and will also delete the object ... delState not

Reply:
Not really ;)

The comment I quoted implies that `delState` removes just the state (not object) but `deleteState` removes *both* the state and the object. You just said the same thing.

If this were true, I should be able to replace this...

```javascript
    try {
        await adapter.delObjectAsync(stateName);
    } catch (err) {
        if (err != 'Error: Not exists') {
            throw new Error(`Failed to delete object ${stateName}: ${err}`);
        }
    }
    try {
        await adapter.delStateAsync(stateName);
    } catch (err) {
        if (err != 'Error: Not exists') {
            throw new Error(`Failed to delete state ${stateName}: ${err}`);
        }
    }
```

... with this...

```javascript
    try {
        await adapter.deleteStateAsync(stateName);
    } catch (err) {
        if (err != 'Error: Not exists') {
            throw new Error(`Failed to delete state ${stateName}: ${err}`);
        }
    }
```

... but that simply doesn't work.

Either I have misunderstood or there is a bug in that code?

Reply:
Please look at the signature !! deleteState normally requires that device, channel and state parts are given separately as parameters ... yes it might also wiork with a full ID but honestly: You EIGHTER work with device/channel/state structures in adapter or with the "full id" ... with the first one you use the create/deleteStates methods, bif other then setObject and delState ... one or the other, do not start to mix please :-)

Reply:
Yeah - I did look at that. And I looked in the code and see that if you only supply one argument it assumes that argument to be the state name and all good.

So what you are saying is that in order to replace two calls to `delObject` & `delState` with a single call to `deleteState` then the adapter must always use device/channel/state in all these calls. OK, if that is what is supposed to happen no worries.

Given that the RPI2 adapter never did that then... well... we just have to leave the two `delObject` & `delState` calls. Ie. this pull request is good as is.

So will you please merge it? :)